opengrok-indexer/src/main/java/org/opengrok/indexer/history/HistoryGuru.java

```diff
@@ -731,24 +729,16 @@ public void ensureHistoryCacheExists(File file) throws HistoryException {
 
         Repository repository = getRepository(file);
         if (repository == null) {
-            // no repository -> no history :(
+            // no repository -> no history
             return;
         }
 
-        String sinceRevision = null;
-
-        if (historyCache.hasCacheForDirectory(file, repository)) {
-            sinceRevision = historyCache.getLatestCachedRevision(repository);
```

Comment:
but what if it has history but for older revision?

Reply:
That would fall through. However I think this is not the place to catch this.